tools/src/wyvern/target/corewyvernIL/expression/Let.java

```diff
@@ -64,4 +65,16 @@ public Value interpret(EvalContext ctx) {
 		Value v = toReplace.interpret(ctx);
 		return inExpr.interpret(ctx.extend(varName, v));
 	}
+
+	@Override
+	public Set<String> getFreeVariables() {
+		
+		// Get free variables in the sub-expressions.
+		Set<String> freeVars = toReplace.getFreeVariables();
+		freeVars.addAll(inExpr.getFreeVariables());
+		
+		// Remove the name that just became bound.
+		freeVars.remove(varName);
```

Comment:
This is not the right semantics - start with the free vars from inExpr, then subtract varName, then add the free vars in toReplace.  That handles the following case correctly: let x = x in x+y [free vars = { x, y}]